Description: Translates JbCacheRedirectorKt.repositories
 This patch translates JbCacheRedirectorKt.repositories with Redirector method
 to work with build.gradle.
Bug-Debian: https://bugs.debian.org/939730

---

--- a/build.gradle
+++ b/build.gradle
@@ -12,13 +12,12 @@
     kotlinBootstrapFrom(BootstrapOption.TeamCity("1.3.30-eap-28", projectExtId = "Kotlin_1330_Compiler", onlySuccessBootstrap = false))
 
 	extra["bootstrapKotlinRepo"] = bootstrapKotlinRepo
-    repositories.withRedirector(project) {
-	//bootstrapKotlinRepo?.let(::maven)
-	/*bootstrapKotlinRepo?.let(::maven) what this does is passes bootstrapkotlinrepo's value into maven i.e maven(bootstrapKotlinrepo)
-        * last known value of bootstrapKotlinRepo is https://teamcity.jetbrains.com/guestAuth/app/rest/builds/buildType:(id:Kotlin_1330_Compiler),number:1.3.30-eap-28,branch:default:any/artifacts/content/maven/ */ 
-       maven(bootstrapKotlinRepo.toString())
-       maven("https://plugins.gradle.org/m2")
-    }
+
+	repositories {
+	        maven({url(bootstrapKotlinRepo.toString())})
+	        maven({url("https://plugins.gradle.org/m2")})
+		}
+    JbCacheRedirectorKt.repositoriesWithRedirector(repositories,project)
 
     // a workaround for kotlin compiler classpath in kotlin project: sometimes gradle substitutes
     // kotlin-stdlib external dependency with local project :kotlin-stdlib in kotlinCompilerClasspath configuration.
--- a/buildSrc/src/main/kotlin/jbCacheRedirector.kt
+++ b/buildSrc/src/main/kotlin/jbCacheRedirector.kt
@@ -55,9 +55,9 @@
 
 fun Project.cacheRedirectorEnabled(): Boolean = findProperty("cacheRedirectorEnabled")?.toString()?.toBoolean() == true
 
-fun RepositoryHandler.withRedirector(project: Project, configuration: RepositoryHandler.() -> Unit) {
+fun repositoriesWithRedirector(repositories: RepositoryHandler, project: Project) {
     configuration()
     if (project.cacheRedirectorEnabled()) {
-        redirect()
+        repositories.redirect()
     }
-}
\ No newline at end of file
+}
